Source/Randomizer.cpp

```diff
@@ -403,11 +403,15 @@ void Randomizer::RandomizeChallenge() {
 void Randomizer::RandomizeAudioLogs() {
     std::vector<int> randomOrder(audiologs.size(), 0);
     std::iota(randomOrder.begin(), randomOrder.end(), 0);
-    Randomize(randomOrder, SWAP::NONE);
+    RandomizeInPlace(randomOrder, SWAP::NONE);
     ReassignNames(audiologs, randomOrder);
 }
 
-void Randomizer::Randomize(std::vector<int>& panels, int flags) {
+void Randomizer::Randomize(std::vector<int> panels, int flags) {
+    return RandomizeInPlace(panels, flags);
+}
```

Comment:
"Randomize" doesn't convey "I'm making a copy" (nor does "in-place" really convey "I'm not making a copy").
I think the more correct solution is to switch Randomize() to pass-by-value, and then use `std::move` whenever you don't care about modifying the vector. This conveys the function's intent of "I do not wish to modify this data" but allows functions to elide a copy when they know it's safe.

Reply:
Err, hmm, that's not right. Well. Let me think about this some more.